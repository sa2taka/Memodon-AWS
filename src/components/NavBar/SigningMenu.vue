<template>
  <nav-bar-menu>
    <v-list-item @click="">
      <v-list-item-avatar>
        <v-img :src="user.iconUrl"></v-img>
      </v-list-item-avatar>

      <v-list-item-content>
        <v-list-item-title v-html="user.displayName"></v-list-item-title>
        <v-list-item-subtitle v-html="user.userName"></v-list-item-subtitle>
      </v-list-item-content>
    </v-list-item>
    <v-list-item>
      <v-list-item-content>
        <v-list-item-title>
          <div class="signout-button">
            <v-btn depressed rounded color="primary" @click="signout">
              <v-icon small left>fa-sign-out-alt</v-icon>
              <p class="my-auto">Signout</p>
            </v-btn>
          </div>
        </v-list-item-title>
      </v-list-item-content>
    </v-list-item>
    <v-divider></v-divider>
    <v-list-item>
      <template v-slot:default="{ active, toggle }">
        <v-list-item-content>
          <v-list-item-title>Dark Mode</v-list-item-title>
        </v-list-item-content>

        <v-list-item-action>
          <dark-theme-switch></dark-theme-switch>
        </v-list-item-action>
      </template>
    </v-list-item>
  </nav-bar-menu>
</template>

<script lang="ts">
import { Component, Watch, Vue } from 'vue-property-decorator';
import { Auth } from 'aws-amplify';

import NavBarMenu from '@/components/NavBar/NavBarMenu.vue';
import DarkThemeSwitch from '@/components/Molecules/darkThemeSwitch.vue';

import theme from '@/store/modules/theme';
import User, { UserState } from '@/store/modules/user';

@Component({
  components: {
    NavBarMenu,
    DarkThemeSwitch,
  },
})
export default class SigningMenu extends Vue {
  private user: UserState = User;

  public created() {
    this.$store.subscribe((mutation, state) => {
      if (mutation.type.startsWith('user/')) {
        this.user = mutation.payload;
      }
    });
  }
  private signout() {
    User.signOut();
    Auth.signOut()
      .then((data) => {
        this.$router.push('/');
      })
      .catch((err) => {
        this.$router.push('/');
      });
  }
}
</script>

<style lang="scss" scoped>
.signout-button {
  position: relative;
  width: 100%;
  text-align: end;
}
</style>
