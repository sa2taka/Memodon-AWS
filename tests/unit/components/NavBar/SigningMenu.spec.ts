import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import Vuex from 'vuex';
import SigningMenu from '@/components/NavBar/SigningMenu.vue';

const localVue = createLocalVue();

localVue.use(Vuetify);

describe('Component', () => {
  let wrapper;

  beforeEach(() => {
    const store = new Vuex.Store({});
    wrapper = shallowMount(SigningMenu, { store, localVue });
  });

  test('is a Vue instance', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
