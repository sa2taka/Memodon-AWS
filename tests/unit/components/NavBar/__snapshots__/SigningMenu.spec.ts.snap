// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is a Vue instance 1`] = `
<nav-bar-menu-stub>
  <v-list-item-stub
    activeclass=""
    tag="div"
  >
    <v-list-item-avatar-stub
      size="40"
    >
      <v-img-stub
        options="[object Object]"
        position="center center"
        src=""
        transition="fade-transition"
      />
    </v-list-item-avatar-stub>
     
    <v-list-item-content-stub>
      <v-list-item-title-stub />
       
      <v-list-item-subtitle-stub />
    </v-list-item-content-stub>
  </v-list-item-stub>
   
  <v-list-item-stub
    activeclass=""
    tag="div"
  >
    <v-list-item-content-stub>
      <v-list-item-title-stub>
        <div
          class="signout-button"
        >
          <v-btn-stub
            activeclass=""
            color="primary"
            depressed="true"
            rounded="true"
            tag="button"
            type="button"
          >
            <v-icon-stub
              left=""
              small=""
            >
              fa-sign-out-alt
            </v-icon-stub>
             
            <p
              class="my-auto"
            >
              Signout
            </p>
          </v-btn-stub>
        </div>
      </v-list-item-title-stub>
    </v-list-item-content-stub>
  </v-list-item-stub>
   
  <v-divider-stub />
   
  <v-list-item-stub
    activeclass=""
    tag="div"
  />
</nav-bar-menu-stub>
`;
